### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by vincenzobaz**

### CODE REVIEW

**Comment by vincenzobaz** [comment]
Thank you @adpi2 for the advice!

**Comment by vincenzobaz** [comment]
Ideas for the next steps:

 - Add this repo to Scala Steward
 - Add scalafmt and scalafix checks

**Comment by vincenzobaz** [comment]
Is it possible to have mdoc fail on warnings? (https://github.com/scalacenter/effective-scala-slide-code/runs/5129243682?check_suite_focus=true#step:6:65)

**Review by julienrf — APPROVED** [review]
Thank you Vincenzo, this will be very helpful!

Could you please update the readme to explain that the root build definition is supposed to be used for development and CI purpose only, and that each project has its own build definition, this is why we have files `project/build.properties` in the root directory, _and_ in each subproject directory? (which is not a normal sbt setup)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: Test code snippets
+on: push
```
Comment (julienrf) [inline .github/workflows/ci.yml:2]: Actually that doesn’t run on PRs, could you fix this, please?

**Pull request merged**